bin/argo-api-authn-scripts/ams-create-users-cloud-info.py

```diff
@@ -0,0 +1,420 @@
+                "https://" + ams_host + "/v1/users/" + user_binding_name +
+                "?key=" + ams_token,
+                data=json.dumps(usr_create), verify=verify)
+            LOGGER.info(ams_usr_crt_req.text)
+
+            # if the response doesn't contain the field uuid,
+            # it means the user was not created
+            req_data = json.loads(ams_usr_crt_req.text)
+            if ams_usr_crt_req.status_code != 200:
+                LOGGER.critical("\nUser: " + user_binding_name)
+                LOGGER.critical(
+                    "\nSomething went wrong while creating ams user." +
+                    "\nBody data: " + str(usr_create) + "\nResponse Body: " +
+                    ams_usr_crt_req.text)
+            if "uuid" not in req_data:
+                LOGGER.critical("uuid field not found in response from ams")
+                continue
+
+            # Create the respective AUTH binding
+            bd_data = {
+                'name': user_binding_name,
+                'service_uuid': authn_service_uuid,
+                'host': authn_service_host,
+                'dn': service_dn,
+                'unique_key': req_data["uuid"]
+            }
+            authn_binding_crt_req = requests.post(
+                "https://"+authn_host+"/v1/bindings?key="+authn_token,
+                data=json.dumps(bd_data), verify=verify)
+            LOGGER.info(authn_binding_crt_req.text)
+
+            if authn_binding_crt_req.status_code != 201:
+                LOGGER.critical(
+                    "Something went wrong while creating a binding." +
+                    "\nBody data: " + str(bd_data) + "\nResponse: " +
+                    authn_binding_crt_req.text)
+                # delete the ams user, since binding could not be created
+                requests.delete(
+                    "https://" + ams_host + "/v1/users/" + user_binding_name +
+                    "?key=" + ams_token, verify=verify)
+                continue
+
+            # if both the user and binding have been created,
+            # assign the user to the acl of the topic
+            services[service_type].append(user_binding_name)
+
+            # count how many users have been created
+            user_count += 1
+
+            # create new topic
+            primary_key = service_endpoint. \
+                find("PRIMARY_KEY").text.replace(' ', '')
+            topic_name = 'SITE_' + sitename + '_ENDPOINT_' + primary_key
+            topic_crt_req = requests.put(
+                "https://" + ams_host + "/v1/projects/" + ams_project +
+                "/topics/" + topic_name + "?key=" + ams_token, verify=verify)
+            if topic_crt_req.status_code != 200:
+                LOGGER.critical(
+                    "Something went wrong while creating topic " +
+                    topic_name + "\nResponse: " + topic_crt_req.text)
+                continue
+
+            # modify the authorized users
```

Comment:
Since the script is an 1-1 matching between user and topic, do we need the following functionality, also found in the ams-goc-db-users script?

Reply:
Removed redundant functionality.